magicicadaprotocol/request.py

```diff
@@ -101,7 +101,7 @@ def get_new_request_id(self):
     def connectionLost(self, reason=connectionDone):
         """Abort any outstanding requests when we lose our connection."""
         Protocol.connectionLost(self, reason)
-        requests = self.requests.values()
+        requests = list(self.requests.values())  # make a copy
```

Comment:
why copy this here?

Reply:
The for loop that comes right after this changes self.requests in a way that if we don't make a copy, we get this error:

```
    for request in requests:
builtins.RuntimeError: dictionary changed size during iteration
```